course3/homework-transcoder/transcoder.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <string>
+#include <boost/property_tree/ptree.hpp>
+
+
+namespace Karun
+{
+
+class Transcoder
+{
+public:
+	Transcoder() = default;
+	Transcoder(const std::string& fileName);
+	
+	void ReadXML(const std::string& fileName);
+	void ReadJSON(const std::string& fileName);
+	void WriteXML(const std::string& fileName);
+	void WriteJSON(const std::string& fileName);
+	
+private:
+	boost::property_tree::ptree mData;
```

Comment:
Does it really have to be a class? There is no much state in it, and this could be much simple implemented as a pair of functions